### ISSUE TITLE
[BUG]  Metric 'min_val_loss' is malformed; persisted metric data contained 1 fields. Expected 2 or 3 fields.

### ISSUE THREAD

**Issue by hoangphucITJP** [mlflow/mlflow#3052]
**Please fill in this bug report template to ensure a timely and thorough response.**

### Willingness to contribute
- [ ] Yes. I can contribute a fix for this bug independently.
- [x] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.
- [ ] No. I cannot contribute a bug fix at this time.

### System information
- **OS Platform and Distribution: Ubuntu 18.04
- **MLflow installed from: binary
- **MLflow version: 1.9.1
- **Python version**: 3.6.8
- **Exact command to reproduce**:`mlflow server`

### Describe the problem
It says `Metric 'min_val_loss' is malformed; persisted metric data contained 1 fields. Expected 2 or 3 fields.` and No runs are showed.
![image](https://user-images.githubusercontent.com/14938527/86527397-9b82d200-bec8-11ea-80e8-c29becfe44c1.png)

### Code to reproduce issue
Interrupt the mlflow server process while running experiment then restart it.

### What component(s), interfaces, languages, and integrations does this bug affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [x] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations

**Comment by hoangphucITJP** [comment]
It works now after I removed the last line in metric logs:
```
\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00
```

**Comment by astronauti** [comment]
I am having the same problem. I am using the autolog function for keras and logging the training and validation loss

**Comment by fernandorovai** [comment]
Any hint to help debugging and finding in which file the problem is?
This error happens to me very often and it is really hard to find the problematic file.

**Comment by ARRRRGH** [comment]
Are there log files or is there an other option to get more verbose error messages / the full stack trace. I have the same problem as @fernandorovai : I need to fix the erroneous file by hand, but I can't which one is causing problems.

**Comment by mrbaloglu** [comment]
I had a similar issue when training and logging a PyTorch model. In the line when I log the metric: 
`mlflow.log_metric("validation_loss", loss)` the data type of loss was a torch.Tensor. When I changed it into a numpy data type, the was resolved for me.

**Comment by marquis08** [comment]
I wish the error message would tell me which run to trace. Currently, I have to check every runs in certain experiment one by one. Or is there any smart way ?

**Comment by halqadasi** [comment]
![image](https://github.com/mlflow/mlflow/assets/98920963/5748a8d2-16f6-4e7b-b72e-da1e7ac21557)
I removed the last line, the problem has been solved!

**Comment by HessamLa** [comment]
I've had the exact same issue. The problem arises when the mlflow server shuts down during a metric log session (electrical problems for me). As a result, some of the metric files turn out to be corrupt. In some cases, the file ends with a trail of '\00' bytes. In other cases, the last row of the metric file is malformed. In both cases, the remedy would be to remove the corrupt line from the end of the file.

In my case, I had more than 1000 corrupt metric files. I had to write a script to find the corrupt files and fix the last line.

Finally, I believe it would be a good option if the mlflow server could keep track of the files that are opened and being written. This way, we can find the files that might be corrupt.
